src/meep.hpp

```diff
@@ -1488,31 +1522,52 @@ class fields {
   dft_flux add_dft_flux(const volume_list *where,
 			double freq_min, double freq_max, int Nfreq);
 
+  dft_fields add_dft_fields(component *components, int num_components,
+                            const volume where,
+                            double freq_min, double freq_max, int Nfreq);
+  dft_fields add_dft_fields(int *components, int num_components,
```

Comment:
If this override is only useful for python, you could add it to `meep.i` using the [extend](http://www.swig.org/Doc3.0/SWIGDocumentation.html#Python_nn43) directive instead of including it in the library.

The other option, since `num_components` will never be very big (right?), would be to accept a list instead of a numpy array.  Then the typemaps would look like this:

```
%typemap(in) (meep::component *components, int num_components) {
    if (!PyList_Check($input)) {
        PyErr_SetString(PyExc_ValueError, "Expected a list");
        SWIG_fail;
    }
    $2 = PyList_Size($input);
    $1 = new meep::component[$2];

    for (Py_ssize_t i = 0; i < $2; i++) {
        $1[i] = (meep::component)PyInteger_AsLong(PyList_GetItem(cs, i));
    }
}

%typemap(freearg) (meep::component *components, int num_components) {
    delete[] $1;
}
```

Reply:
Thanks! I put this into `meep.i.` I changed `cs` to `$input` in the line containing `PyInteger_AsLong.`